File: Lab01/Python_Lang/Demo09.py
#
#  Created by NDHuy at 11/04/2020
#  Copyright 2020 by SEstudio
#  Enter one day (day, month, year)
#  Calculate and print the next day
#

def main():
    day = int(input("Enter day: "))
    month = int(input("Enter month: "))
    year = int(input("Enter year: "))

    if (0 < day and day <= 31) and (0 < month and month <= 12) and (0 < year):
        if month == 12:
            if day == 31:
                day = 1
                month = 1
                year += 1
            else:
                day += 1
        elif month == 2:
            if day < 28:
                day += 1
            else:
                if day == 28:
                    if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
                        day = 29
                    else:
                        day = 1
                        month += 1
                else:
                    day = 1
                    month += 1
        elif (month == 1) or (month == 3) or (month == 5) or (month == 7) or (month == 8) or (month == 10):
            if day == 31:
                day = 1
                month += 1
            else:
                day += 1
        elif (month == 4) or (month == 6) or (month == 9) or (month == 11):
            if day == 30:
                day = 1
                month += 1
            else:
                day += 1
        
        print("The next day %d/%d/%d" %(day, month, year))

    else:
        print("Invalid date %d/%d/%d" %(day, month, year))



if __name__ == "__main__":
    main()

File: Lab01/C_Lang/src/Demo09.cpp
//
//  Created by NDHuy at 11/04/2020
//  Copyright 2020 by SEstudio
//  Enter one day (day, month, year)
//  Calculate and print the next day
//

#include <stdio.h>

int main()
{
    int day; 
    int month; 
    int year;

    printf("Enter day: ");
    scanf("%d", &day);

    printf("Enter month: ");
    scanf("%d", &month);

    printf("Enter year: ");
    scanf("%d", &year);

    // Check input
    if((day > 0 && day <= 31) && (month > 0 && month < 13) && year > 0)
    {

        switch (month)
        {
            case 2:
                // Check Leap year
                if(((year % 4 == 0) && (year % 100 != 0)) || (year % 400 == 0))
                {
                    if(day == 28)
                    {
                        day = 29;
                    }
                    else
                    {
                        if(day > 28)
                        {
                            day = 1;
                            month++;
                        }
                        else
                        {
                            day++;
                        }
                    }
                }
                else
                {
                    if(day == 28)
                    {
                        day = 1;
                        month++;
                    }
                    else
                    {
                        day++;
                    }
                    
                }
                break;

            case 1:
            case 3:
            case 5:
            case 7:
            case 8:
            case 10:
                if(day == 31)
                {
                    day = 1;
                    month++;
                }
                else
                {
                    day++;
                }
                break;

            case 4:
            case 6:
            case 9:
            case 11:
                if(day == 30)
                {
                    day = 1; 
                    month++;
                }
                else
                {
                    day++;
                }
                
                break;

            case 12:
                if(day == 31)
                {
                    day = 1;
                    month = 1;
                    year = year + 1;
                }
                else
                {
                    day++;
                }
                break;
            
            default:
                break;
        }
        
        printf("The next day: %d/%d/%d \n", day, month, year);
    }
    else
    {
        printf("Invalid date %d/%d/%d \n", day, month, year);
    }

    return 0;
}